oeps/oep-0013.rst

```diff
@@ -0,0 +1,576 @@
+
+There can be three levels for a related object (using ``username`` as an example):
+
+    (a) Returning just the id::
+
+            "username": "andya"
+
+    (b) Returning an object containing the id and a URL to get more details::
+
+            "user": {
+              "username": "andya",
+              "url": "https://openedx.example.com/api/user/v1/accounts/andya"
+            }
+
+    (c) Returning an object containing the expanded details::
+
+            "user": {
+              "username": "andya",
+              "country": "UK",
+              "profile_image": {
+                "has_image": True,
+                "image_url_full": "http://my-image-storage.net/media/profile_images/123456789_500.jpg",
+                "image_url_large": "http://my-image-storage.net/media/profile_images/123456789_120.jpg",
+                "image_url_medium": "http://my-image-storage.net/media/profile_images/123456789_50.jpg",
+                "image_url_small": "http://my-image-storage.net/media/profile_images/123456789_30.jpg",
+              },
+              ...
+            }
+
+By default:
+
+- For objects that have their own APIs, use option (b) and return both an id and a URL
+- For other objects, use option (a) and just return the id
+- Support a query parameter indicating an optional list of objects to be expanded: e.g. ``?expand=user,team``
+
+
+Review process
+--------------
+
+At the moment, there is no formal review process for new APIs. Please work with the relevant maintainer team to determine if API changes make sense for your use case.
+
+Rationale
+=========
+
+Our conventions were originally derived from a very useful set defined by `Apigee <http://pages.apigee.com/rs/apigee/images/api-design-ebook-2012-03.pdf>`_. Portions of that guide have been reused here.
+
+Open edX is not the first project that has attempted to define API best practices. Here are some other organizations that have formalized their API practices:
+
+- `Google <https://developers.google.com/+/api/>`_
+- `Facebook <https://developers.facebook.com/docs/graph-api/reference/v2.2/profile>`_
+- `Twitter <https://dev.twitter.com/rest/public>`_
+- `LinkedIn <https://developer.linkedin.com/docs/rest-api>`_
+- `StackExchange <https://api.stackexchange.com/>`_
+- `Instagram <https://instagram.com/developer/>`_
+- `Github <https://developer.github.com/v3/>`_
+- `Salesforce <https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/intro_rest_resources.htm>`_
+
+
+Backward Compatibility
+=======================
+
+Any web APIs that do not conform with these practices should be updated accordingly. This document is based on existing practices so we do not expect any new backwards compatibility issues.
+
+
+Reference Implementation
+========================
+
+There are a number of existing APIs that you can crib from, for example:
```

Comment:
It would be good to update this to include some of the newer APIs, such as those for course discovery.